src/nxdoc/nuxeo-server/administration/maintenance/garbage-collecting-orphaned-blobs.md

```diff
@@ -0,0 +1,170 @@
+---
+title: Garbage-Collecting Orphaned Blobs
+review:
+    comment: ''
+    date: '2023-06-09'
+    status: ok
+labels:
+    - http
+    - rest-api
+toc: true
+tree_item_index: 999
+---
+
+## Immediate Garbage Collection
+
+Since [LTS 2021-HF35]({{page page='nuxeo-server-release-notes-2021-35'}}) (see [NXP-31594](https://jira.nuxeo.com/browse/NXP-31594)), the Nuxeo Platform deletes orphaned blobs whenever a:
+- document is removed
+- document blob property is edited
+- document blob property is dispatched to another blob provider
+
+### Preconditions
+
+Only deployments using the **MongoDB** backend can benefit from this feature. All the following conditions must also be met:
+
+#### Repository with `queryBlobKeys` capability
+
+This new GC implementation only works for repositories having the `queryBlobKeys` capability.
+
+Since LTS 2021-HF02 and [NXP-29516](https://jira.nuxeo.com/browse/NXP-29516), the blob keys referenced by a document are stored in its `ecm:blobKeys` field.
+
+If **ALL** documents of a repository have this field computed, then the repository has the `queryBlobKeys` capability. In other words, a repository with documents created by a nuxeo server with a version prior to LTS 2021.2 / LTS 2021-HF02 **does NOT** have this capability.
+
+You can query the [capability endpoint]({{page page='rest-api-endpoints#capability-endpoint'}}) to check whether a repository has the `queryBlobKeys` capability.
+
+In case of multi-repository configuration, all the repositories must have this capability.
+
+##### Acquire the `queryBlobKeys` capability
+
+In the case of missing `queryBlobKeys` capability in your repository, the `blob-keys-migration` migration is available to acquire it since LTS 2021-HF39. This migration process goes through all the documents of the repository to compute their `ecm:blobKeys` field value. Depending on the volume of your data, this may be a long-running process.
+
+###### Check repository does not have the capability
+
+```curl
+curl -u Administrator:Administrator http://localhost:8080/nuxeo/api/v1/capabilities
+```
+
+```json
+{
+   "cluster" : {
+      "enabled" : false,
+      "nodeId" : "1231481178147993942"
+   },
+   "entity-type" : "capabilities",
+   "repository" : {
+      "default" : {
+         "queryBlobKeys" : false
+      }
+   },
+   "server" : {
+      "distributionName" : "lts",
+      "distributionServer" : "tomcat",
+      "distributionVersion" : "2021.39.3"
+   }
+}
+```
+Observe `repository.default.queryBlobKeys` equals **false**
+
+###### Run the `blob-keys-migration` migration
+
+```curl
+curl -X POST -u Administrator:Administrator http://localhost:8080/nuxeo/api/v1/management/migration/blob-keys-migration/run
+```
+No output expected.
+
+###### Wait for the migration
```

Comment:
As an IT manager, how can I tell if something went wrong and debug?

Reply:
look at the logs

Reply:
It is indeed relying on a BAF. If any failure, it will stop and the capability will not be acquired.

>how do I know which documents are missing

We don't, we need to go through all docs which have a `NULL` `ecm:blobKeys` field

>It might also be worth considering they may be importing content during the process => would it be covered, or do they need another run for it?

As long as you are running a server >= `LTS 2021-HF02`, the `ecm:blobKeys` is computed whenever a doc is created/updated even while the migration s running.

I can somehow add this in the page